### PR TITLE
Enhance qa_automation/qa_run.pm for easier debugging

### PR DESCRIPTION
**Pull request by jtzhao**
Besides /var/log/qaset/log, also compress and upload calls/control/runs/set/submission directories.

### CODE REVIEW

**Comment by okurz** [comment]
:+1:

**Comment by jtzhao** [comment]
No idea why the travis check failed since "tidy --check" didn't say anything on my laptop.
Anyway, it's fixed now.

File: tests/qa_automation/qa_run.pm [hunk 1]
```diff
@@ -62,9 +63,13 @@ sub prepare_repos {
         assert_script_run($rm_repos, 300);
         assert_script_run("zypper --no-gpg-check -n ar -f '$qa_server_repo' server-repo");
     }
-    assert_script_run "zypper --no-gpg-check -n ar -f '" . get_var('QA_HEAD_REPO') . "' qa-ibs";
+    my $qa_head_repo = get_var('QA_HEAD_REPO', '');
+    unless ($qa_head_repo) {
+        die "No QA_HEAD_REPO specified!";
```
Comment (coolo) [inline tests/qa_automation/qa_run.pm]: btw: you can use get_required_var instead

**Comment by coolo** [comment]
https://openqa.suse.de/tests/271977 is yours

**Comment by jtzhao** [comment]
@coolo Thanks. Looks like it's working properly.

**Pull request merged**